### PR TITLE
Cyril/refine single sample upload design

### PR DESCRIPTION
**Pull request by cyrielo**
# Description
This PR incorporates the new design layout for uploading a single sample

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Impacted Areas in Application
List general components of the application that this PR will affect:

Which pages?
- [x] Single Upload Page

Specific components 

* SampleUpload

# Testing Script:

### Single Upload
* Using an AWS path that doesn't exist throws an error
* Successfully uploaded files from a folder (use test host)
* Uploading a duplicate file throws an error
* Pipeline gets initiated successfully 

# Checklist:

- [x] I have run the testing script to make sure current functionality is unchanged
- [x] I have done relevant tests that prove my fix is effective or that my feature works
- [x] I have spent time testing out edge cases for my feature
- [x] I have updated the test script or pull request template if necessary
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/assets/javascripts/components/sampleUpload.jsx [hunk 1]
```diff
@@ -51,18 +51,20 @@ class SampleUpload extends React.Component {
       selectedJobQueue: this.selected.jobQueue || '',
```
Comment (jameshowardwang) [inline app/assets/javascripts/components/sampleUpload.jsx]: i know this wasn't part of this diff review but in line 23 of this file, there is a reference to hostGenomes[0] which breaks if there aren't any hostGenomes uploaded yet (e.g. when I do a ./bin/setup from scratch). could you please fix that, either as part of this diff, or separately?

Comment (cyrielo) [reply app/assets/javascripts/components/sampleUpload.jsx]: Sure would reproduce this error and apply a fix

File: app/assets/javascripts/components/sampleUpload.jsx [hunk 2]
```diff
@@ -267,106 +263,145 @@ class SampleUpload extends React.Component {
...
-        errorMessage: 'Please select a host genome'
-      })
-      return true;
+    const errors = {};
+    if(this.state.selectedProject) {
+      if(this.state.selectedProject.toLowerCase() === 'select a project') {
+        errors.selectedProject = 'Please select a project';
+      }
+    } else {
+      errors.selectedProject = 'Please select a project';
```
Comment (cdebourcy) [inline app/assets/javascripts/components/sampleUpload.jsx]: Could you make text like 'select a project' and 'Please select a project' constants so that we only have to change them in the code once if we want to modify them?

File: app/assets/javascripts/components/sampleUpload.jsx [hunk 3]
```diff
@@ -127,7 +131,7 @@ class SampleUpload extends React.Component {
     .catch((error) => {
       that.setState({
         invalid: true,
-        errorMessage: 'Project exists already or is invalid',
+        errors: { 'selectedProject': 'Project already exists or is in-valid' },
```
Comment (cdebourcy) [inline app/assets/javascripts/components/sampleUpload.jsx]: I think it should be `invalid` rather than `in-valid`

**Comment by cdebourcy** [comment]
It doesn't seem to quite work yet:
screen shot 2017-12-04 at 10 48 13 am

File: app/assets/javascripts/components/sampleUpload.jsx (hunk 1 above)
Comment (cdebourcy) [reply app/assets/javascripts/components/sampleUpload.jsx]: I think the solution would be to seed the database with a host genome as part of bin/setup. I can add that

Comment (cyrielo) [reply app/assets/javascripts/components/sampleUpload.jsx]: Yeah I have also at least prevented the page from breaking if there aren't any host genomes
